src/app/store/chains/stx-chain.actions.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  createWalletGaiaConfig,
+  generateNewAccount,
+  updateWalletConfig,
+  Wallet,
+} from '@stacks/wallet-sdk';
+
+import { gaiaUrl } from '@shared/constants';
+import { logger } from '@shared/logger';
+import { saveWalletConfigLocally } from '@shared/utils/wallet-config-helper';
+import { AppThunk } from '@app/store';
+import { selectDefaultWalletKey } from '../in-memory-key/in-memory-key.selectors';
+import { stxChainSlice } from './stx-chain.slice';
+
+export const createNewAccount = (wallet: Wallet): AppThunk => {
+  return async (dispatch, getState) => {
+    const secretKey = selectDefaultWalletKey(getState());
+    if (!secretKey) {
+      throw new Error('Unable to create a new account - not logged in.');
```

Comment:
Should this be `...not signed in.` bc we use `sign in` and `sign out` typically?